mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -420,203 +277,33 @@ func marshallSearchIndexMappingFields(fields map[string]matlas.IndexField) (stri
-	mapCharFilters := make([]map[string]interface{}, len(filters))
-
-	for i, filter := range filters {
-		mapCharFilters[i] = map[string]interface{}{
-			"type": filter.Type,
-		}
-
-		if filter.OriginalTokens != "" {
-			mapCharFilters[i]["original_tokens"] = filter.OriginalTokens
-		}
-		//
-
-		if filter.Min != nil {
-			mapCharFilters[i]["min"] = *filter.Min
-		}
-
-		if filter.Max != nil {
-			mapCharFilters[i]["max"] = *filter.Max
-		}
-
-		if filter.NormalizationForm != "" {
-			mapCharFilters[i]["normalization_form"] = filter.NormalizationForm
-		}
-
-		if filter.MinGram != nil {
-			mapCharFilters[i]["min_gram"] = *filter.MinGram
-		}
-
-		if filter.MaxGram != nil {
-			mapCharFilters[i]["max_gram"] = *filter.MaxGram
-		}
-
-		if filter.TermsNotInBounds != "" {
-			mapCharFilters[i]["terms_not_in_bounds"] = filter.TermsNotInBounds
-		}
-
-		if filter.MinShingleSize != nil {
-			mapCharFilters[i]["min_shingle_size"] = *filter.MinShingleSize
-		}
-
-		if filter.MaxShingleSize != nil {
-			mapCharFilters[i]["max_shingle_size"] = *filter.MaxShingleSize
-		}
-
-		if filter.Pattern != "" {
-			mapCharFilters[i]["pattern"] = filter.Pattern
-		}
-
-		if filter.Replacement != "" {
-			mapCharFilters[i]["replacement"] = filter.Replacement
-		}
-
-		if filter.Matches != "" {
-			mapCharFilters[i]["matches"] = filter.Matches
-		}
-
-		if filter.StemmerName != "" {
-			mapCharFilters[i]["stemmer_name"] = filter.StemmerName
-		}
-
-		if len(filter.Tokens) > 0 {
-			mapCharFilters[i]["tokens"] = filter.Tokens
-		}
-
-		if filter.IgnoreCase != nil {
-			mapCharFilters[i]["ignore_case"] = *filter.IgnoreCase
-		}
-	}
-	return mapCharFilters
-}
-
-func flattenSearchIndexCharFilters(filters []*matlas.AnalyzerCharFilter) ([]map[string]interface{}, error) {
-	if len(filters) == 0 {
-		return nil, nil
-	}
-
-	mapCharFilters := make([]map[string]interface{}, len(filters))
-
-	for i, filter := range filters {
-		mapCharFilters[i] = map[string]interface{}{
-			"type": filter.Type,
-		}
-
-		if len(filter.IgnoreTags) > 0 {
-			mapCharFilters[i]["ignore_tags"] = filter.IgnoreTags
-		}
-
-		if filter.Mappings != nil {
-			searchIndexCharFilterMappingFields, err := marshallSearchIndexCharFilterMappingFields(*filter.Mappings)
-			if err != nil {
-				return nil, err
-			}
-
-			mapCharFilters[i]["mappings"] = searchIndexCharFilterMappingFields
-		}
-	}
-	return mapCharFilters, nil
-}
-
 func resourceMongoDBAtlasSearchIndexCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	// Get client connection.
 	conn := meta.(*MongoDBClient).Atlas
 	projectID := d.Get("project_id").(string)
 
 	clusterName := d.Get("cluster_name").(string)
 
-	var indexMapping *map[string]matlas.IndexField
-	indexMappingResult := unmarshalSearchIndexMappingFields(d.Get("mappings_fields").(string))
-	indexMapping = &indexMappingResult
+	var indexMapping map[string]interface{}
+	indexMapping = unmarshalSearchIndexMappingFields(d.Get("mappings_fields").(string))
```

Comment:
NIT: combine the last two lines into one.

Reply:
this was already changed